### PR TITLE
Accept config file as another option to inject configs

### PR DESCRIPTION
**Pull request by yee2542**
add new args `--config=./sample/config/config.yml` for config via YAML format file.
when using the config file, the argument in cli will be ignored.

example for the config file

```yml
provider:
  url: https://github.com/codeleague/codecoach
  pr: 40
  token: 123qwe123qwe123qwe123
buildLogFiles:
  - type: tslint
    path: /path/to/log.json
    cwd: /repo/src
output: ./output.json

```

### CODE REVIEW

File: src/Config/configBuilder.ts [hunk 1]
```diff
@@ -1,18 +1,32 @@
...
+): Promise<ProviderConfig> => {
+  const configFile = await buildYMLConfig(arg);
+
+  return {
+    token: configFile?.provider.token || arg.token,
+    repoUrl: configFile?.provider.url || arg.url,
+    prId: configFile?.provider.pr || arg.pr,
+    removeOldComment: configFile?.provider.removeOldComment || arg.removeOldComment,
+    userAgent: configFile?.provider.userAgent || USER_AGENT,
+    timeZone: configFile?.provider.timeZone || TIME_ZONE,
```
Comment (encX) [inline src/Config/configBuilder.ts:22]: again, this might not needed in config file as it's already hardcoded.

File: src/Config/configBuilder.ts [hunk 2]
```diff
@@ -1,18 +1,32 @@
...
-  repoUrl: arg.url,
-  prId: arg.pr,
-  removeOldComment: arg.removeOldComment,
-  userAgent: USER_AGENT,
-  timeZone: TIME_ZONE,
-});
+const buildYMLConfig = async (args: ConfigArgument) => {
+  if (!args.config) return;
+  const parsed = await YML.parse(args.config);
+  return parsed;
```
Comment (encX) [inline src/Config/configBuilder.ts:8]: This could be inline return await

**Review by encX — CHANGES_REQUESTED** [review]
Don't mad at me na, I review hai leaw 555

Really good job! I left some comments, please check.
And please add documentation about config file usage in README file
_(see [config](https://github.com/codeleague/codecoach#configs) section.)_

File: sample/config/config.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+provider:
+  url: https://github.com/codeleague/codecoach
+  pr: 40
+  token: 123qwe123qwe123qwe123
+  userAgent: CodeCoach
+  timeZone: Asia/Bangkok
```
[the remark this answers is not in the document]
Comment (yee2542) [reply sample/config/config.yaml:6]: GitHubPrService.ts not consume a config 555555555555555555
should able to config mai ? or just leave it and open issue

[the remark this answers is not in the document]
Comment (encX) [reply sample/config/config.yaml]: leave it there for now 55555

**Review by encX — APPROVED** [review]
Thankssss ❤️

**Pull request merged**